src/super_gradients/common/auto_logging/console_logging.py

```diff
@@ -0,0 +1,112 @@
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from super_gradients.common.environment.env_helpers import multi_process_safe, is_distributed
+
+
+class StdoutTee(object):
+    """Duplicate the stdout stream to save it into a given file."""
+
+    def __init__(self, file):
+        self.file = file
+        self.stdout = sys.stdout
+        sys.stdout = self
+
+    def __del__(self):
+        sys.stdout = self.stdout
+        self.file.close()
+
+    def write(self, data):
+        self.file.write(data)
+        self.stdout.write(data)
+
+    def flush(self):
+        self.file.flush()
+
+    def __getattr__(self, attr):
+        return getattr(self.stdout, attr)
+
+
+class StderrTee(object):
+    """Duplicate the stderr stream to save it into a given file."""
+
+    def __init__(self, file):
+        self.file = file
+        self.stderr = sys.stderr
+        sys.stderr = self
+
+    def __del__(self):
+        sys.stderr = self.stderr
+        self.file.close()
+
+    def write(self, data):
+        self.file.write(data)
+        self.stderr.write(data)
+
+    def flush(self):
+        self.file.flush()
+
+    def __getattr__(self, attr):
+        return getattr(self.stderr, attr)
+
+
+def copy_file(src_filename: str, dest_filename: str, copy_mode: str = "w"):
+    """Copy a file from source to destination. Also works when the destination folder does not exist."""
+    os.makedirs(os.path.dirname(dest_filename), exist_ok=True)
+    if os.path.exists(src_filename):
+        with open(src_filename, "r", encoding="utf-8") as src:
+            with open(dest_filename, copy_mode) as dst:
+                dst.write(src.read())
+
+
+class ConsoleSink:
+    """Singleton responsible to sink the console streams (stdout/stderr) into a file."""
+
+    def __init__(self):
+        self._setup()
+
+    @multi_process_safe
+    def _setup(self):
+        """On instantiation, setup the default sink file."""
+        default_path = Path.home() / "sg_logs" / "console.log"
```

Comment:
Pathlib works fine on Windows, so should not be an issue. I think the biggest advantage of Pathlib is this nice syntax sugar to allow writing `path / to / something` instead of os.path.join(path,to,something)`. But yes, it is cross-platform.